transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/CustomAttributesStoreImpl.kt

```diff
@@ -30,10 +40,14 @@ internal class CustomAttributesStoreImpl(
     }
 
     private fun isCustomAttributesValid(customAttributes: Map<String, String>): Boolean {
-        if (customAttributes.isEmpty() || this.customAttributes == customAttributes) {
-            log.w { LogMessages.CUSTOM_ATTRIBUTES_EMPTY_OR_SAME }
-            return false
-        } else if (isSizeExceeded(customAttributes)) {
+        return maxCustomDataBytes.isUnset() || (
+            !(customAttributes.isEmpty() || this.customAttributes == customAttributes) &&
+                !maybeReportFailure(customAttributes)
+            )
+    }
+
+    private fun maybeReportFailure(customAttributes: Map<String, String>): Boolean {
+        return isSizeExceeded(customAttributes).also {
```

Comment:
code inside `also` scope will be executed each and every time leading to report Event.Error on every invocation of  `maybeReportFailure`.